### ISSUE TITLE
Angular 16 error due to ng2-charts-schematics

### ISSUE THREAD

**Issue by sydneyhenrard** [valor-software/ng2-charts#1681]
I migrated from Angular 15.2.0 to Angular 16.1.6 and tried to upgrade from `ng2-charts` 4.1.1 to 5.0.1. I got this error due to `ng2-charts-schematics` when running `npm install`.
```
npm WARN Could not resolve dependency:
npm WARN peer ng2-charts-schematics@"0.0.1" from ng2-charts@5.0.1
npm WARN node_modules/ng2-charts
npm WARN   ng2-charts@"^5.0.1" from the root project
```
package.json
```
"dependencies": {
    "@angular/animations": "^16.1.6",
    "@angular/cdk": "^16.1.5",
    "@angular/common": "^16.1.6",
    "@angular/compiler": "^16.1.6",
    "@angular/core": "^16.1.6",
    "@angular/forms": "^16.1.6",
    "@angular/material": "^16.1.5",
    "@angular/platform-browser": "^16.1.6",
    "@angular/platform-browser-dynamic": "^16.1.6",
    "@angular/router": "^16.1.6",
    "angular-oauth2-oidc": "^15.0.1",
    "chart.js": "^4.3.0",
    "chartjs-adapter-date-fns": "^3.0.0",
    "cross-env": "^7.0.3",
    "date-fns": "^2.30.0",
    "flag-icons": "^6.7.0",
    "ng2-charts": "^5.0.1",
    "ngx-auto-unsubscribe": "^3.0.1",
    "ngx-date-fns": "^10.0.1",
    "ngx-google-analytics": "^14.0.1",
    "rxjs": "~7.5.0",
    "tslib": "^2.3.0",
    "zone.js": "~0.13.1"
  },
```

**Comment by anishchenko-anton** [comment]
I have the same issue

**Comment by uplink42** [comment]
Same error here. I'm unable to install any of the v5 versions:

![image](https://github.com/valor-software/ng2-charts/assets/7904753/9997e030-03e3-4597-90ac-5a1bcc099de4)

**Comment by MikaStark** [comment]
Same issue.

I tried to install it on a blank project and the schematic install version 4 instead of 5. But it worked.
It worth nothing to say that the schematic is not compatible with standalone project.

**Comment by wjehring** [comment]
I'm not sure about the build/npm publish process that's used for this project, but the issue is that the [schematic package.json](https://github.com/valor-software/ng2-charts/blob/387aa8bbd7722615a2470278d3645619b75413d9/libs/ng2-charts-schematics/package.json) references a version that doesn't exist on npm.

When the ng2-charts lib is built, nx automatically generates the dependencies in the package.json using that version number, but it doesn't exist on npm. I imagine updating the schematics version to one that _does_ exist in npm would fix this.

In the meantime, you should be able to fix this in your own project by adding this to your package.json:

```
"overrides": {
  "ng2-charts": {
    "ng2-charts-schematics": "0.1.7"
  }
}
```

**Comment by changhuixu** [comment]
having same issue here.
seems like the `ng2-charts-schematics` is required.

Also, when using `ng add ng2-charts`, it installs `chart.js` v3.6.0 instead of v4

**Comment by maranmaran** [comment]
Thumbs up

**Comment by ekselys** [comment]
same here

**Comment by Muminniyoz** [comment]
same problem

**Comment by FerAnimaciones** [comment]
same problem 🙂

**Comment by SebastianStehle** [comment]
Same here. Do you need support or funding to handle this asap. It is a show blocker for me :(

**Comment by tobigit** [comment]
Same Problem here.

**Comment by SebastianStehle** [comment]
Sorry, I have not seen @wjehring solution ;)

**Comment by ptomaszi** [comment]
Same problem

**Comment by idenisovs** [comment]
I'm not sure, if it is the same, but today, during the `ng add ng2-charts` I got the follwing:

```
Collection "/private/tmp/testable/node_modules/ng2-charts/schematics/src/collection.json" cannot be resolved.
```

It is complete new project, just created with `ng new` command, the Angular version `16.2.0`.

**Comment by dhakerbadri** [comment]
Same problem here !

**Comment by dhakerbadri** [comment]
Thank you @wjehring !

**Comment by wjehring** [comment]
Sounds like people are getting round this issue with the package.json overrides.

If you see any complaints during dependency installation due to `@angular/platform-browser` then you may also need to add [this override](https://github.com/valor-software/ng2-charts/issues/1682#issuecomment-1653541543).

**Comment by santam85** [comment]
This should be fixed in 5.0.3

**Comment by Ahmed-Tombari** [comment]
> I migrated from Angular 15.2.0 to Angular 16.1.6 and tried to upgrade from `ng2-charts` 4.1.1 to 5.0.1. I got this error due to `ng2-charts-schematics` when running `npm install`.
> 
> ```
> npm WARN Could not resolve dependency:
> npm WARN peer ng2-charts-schematics@"0.0.1" from ng2-charts@5.0.1
> npm WARN node_modules/ng2-charts
> npm WARN   ng2-charts@"^5.0.1" from the root project
> ```
> 
> package.json
> 
> ```
> "dependencies": {
>     "@angular/animations": "^16.1.6",
>     "@angular/cdk": "^16.1.5",
>     "@angular/common": "^16.1.6",
>     "@angular/compiler": "^16.1.6",
>     "@angular/core": "^16.1.6",
>     "@angular/forms": "^16.1.6",
>     "@angular/material": "^16.1.5",
>     "@angular/platform-browser": "^16.1.6",
>     "@angular/platform-browser-dynamic": "^16.1.6",
>     "@angular/router": "^16.1.6",
>     "angular-oauth2-oidc": "^15.0.1",
>     "chart.js": "^4.3.0",
>     "chartjs-adapter-date-fns": "^3.0.0",
>     "cross-env": "^7.0.3",
>     "date-fns": "^2.30.0",
>     "flag-icons": "^6.7.0",
>     "ng2-charts": "^5.0.1",
>     "ngx-auto-unsubscribe": "^3.0.1",
>     "ngx-date-fns": "^10.0.1",
>     "ngx-google-analytics": "^14.0.1",
>     "rxjs": "~7.5.0",
>     "tslib": "^2.3.0",
>     "zone.js": "~0.13.1"
>   },
> ```



> I migrated from Angular 15.2.0 to Angular 16.1.6 and tried to upgrade from `ng2-charts` 4.1.1 to 5.0.1. I got this error due to `ng2-charts-schematics` when running `npm install`.
> 
> ```
> npm WARN Could not resolve dependency:
> npm WARN peer ng2-charts-schematics@"0.0.1" from ng2-charts@5.0.1
> npm WARN node_modules/ng2-charts
> npm WARN   ng2-charts@"^5.0.1" from the root project
> ```
> 
> package.json
> 
> ```
> "dependencies": {
>     "@angular/animations": "^16.1.6",
>     "@angular/cdk": "^16.1.5",
>     "@angular/common": "^16.1.6",
>     "@angular/compiler": "^16.1.6",
>     "@angular/core": "^16.1.6",
>     "@angular/forms": "^16.1.6",
>     "@angular/material": "^16.1.5",
>     "@angular/platform-browser": "^16.1.6",
>     "@angular/platform-browser-dynamic": "^16.1.6",
>     "@angular/router": "^16.1.6",
>     "angular-oauth2-oidc": "^15.0.1",
>     "chart.js": "^4.3.0",
>     "chartjs-adapter-date-fns": "^3.0.0",
>     "cross-env": "^7.0.3",
>     "date-fns": "^2.30.0",
>     "flag-icons": "^6.7.0",
>     "ng2-charts": "^5.0.1",
>     "ngx-auto-unsubscribe": "^3.0.1",
>     "ngx-date-fns": "^10.0.1",
>     "ngx-google-analytics": "^14.0.1",
>     "rxjs": "~7.5.0",
>     "tslib": "^2.3.0",
>     "zone.js": "~0.13.1"
>   },
> ```

Try to use
npm uninstall ng2-charts
npm install ng2-charts@5.0.0
